homeassistant/components/mqtt/models.py

```diff
@@ -373,14 +373,14 @@ def __init__(self) -> None:
     def process_write_state_requests(self, msg: MQTTMessage) -> None:
         """Process the write state requests."""
         while self.subscribe_calls:
-            _, entity = self.subscribe_calls.popitem()
+            entity_id, entity = self.subscribe_calls.popitem()
             try:
                 entity.async_write_ha_state()
-            except Exception:
-                _LOGGER.exception(
-                    "Exception raised when updating state of %s, topic: "
-                    "'%s' with payload: %s",
-                    entity.entity_id,
+            except Exception as exc:  # noqa: BLE001
```

Comment:
If we do a broad exception catch, I think `_LOGGER.exception` is generally preferred since we aren't expecting an exception to be thrown under normal circumstances and it would be a bug here for it to raise during state write.